packages/twenty-front/src/modules/databases/hooks/useGetDatabaseConnection.ts

```diff
@@ -0,0 +1,47 @@
+import { useQuery } from '@apollo/client';
+
+import { GET_ONE_DATABASE_CONNECTION } from '@/databases/graphql/queries/findOneDatabaseConnection';
+import { getForeignDataWrapperType } from '@/databases/utils/getForeignDataWrapperType';
+import { useApolloMetadataClient } from '@/object-metadata/hooks/useApolloMetadataClient';
+import {
+  GetOneDatabaseConnectionQuery,
+  GetOneDatabaseConnectionQueryVariables,
+} from '~/generated-metadata/graphql';
+
+type UseGetDatabaseConnectionParams = {
+  databaseKey: string;
+  connectionId: string;
+  skip?: boolean;
+};
+
+export const useGetDatabaseConnection = ({
+  databaseKey,
+  connectionId,
+  skip,
+}: UseGetDatabaseConnectionParams) => {
+  const apolloMetadataClient = useApolloMetadataClient();
+  const foreignDataWrapperType = getForeignDataWrapperType(databaseKey);
+
+  const { data, loading } = useQuery<
+    GetOneDatabaseConnectionQuery,
+    GetOneDatabaseConnectionQueryVariables
+  >(GET_ONE_DATABASE_CONNECTION, {
+    client: apolloMetadataClient ?? undefined,
+    skip: skip || !apolloMetadataClient || !foreignDataWrapperType,
+    variables: {
+      input: {
+        id: connectionId,
+      },
+    },
+  });
+
+  const connection = data?.findOneRemoteServerById ?? null;
+
+  return {
+    connection:
+      connection?.foreignDataWrapperType === foreignDataWrapperType
```

Comment:
We need to check this somewhere, as both the `databaseKey` and `connectionId` come from the url parameters. If I try to access `postgresql/{airTableConnectionId}`, it should redirect to "not found". Would you rather check this outside this hook?

Reply:
yep I would keep the hook as simple as possible. If we need to check the type because of how the parent component get the params, let's keep the parent validate the data indeed 👍

Reply:
let's merge like this and I'll see if I update it